app/models/notification_request.rb

```diff
@@ -47,6 +47,6 @@ def reconsider_existence!
   private
 
   def prepare_notifications_count
-    self.notifications_count = Notification.where(account: account, from_account: from_account, filtered: true).limit(MAX_MEANINGFUL_COUNT).count
+    self.notifications_count = Notification.where(account_id: account_id, from_account_id: from_account_id, filtered: true).limit(MAX_MEANINGFUL_COUNT).count
```

Comment:
Since this is being called as part of an `after_destroy` hook on notifications I noticed that it wasted a lot of time loading full account records. This change makes the hook considerably faster.